apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/monetary/service/AssetDeleteServiceImpl.java

```diff
@@ -0,0 +1,88 @@
+import com.apollocurrency.aplwallet.apl.core.db.DbIterator;
+import com.apollocurrency.aplwallet.apl.core.db.service.BlockChainInfoService;
+import com.apollocurrency.aplwallet.apl.core.monetary.model.AssetDelete;
+import com.apollocurrency.aplwallet.apl.core.monetary.dao.AssetDeleteTable;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+@Singleton
+public class AssetDeleteServiceImpl implements AssetDeleteService {
+
+    private final AssetDeleteTable assetDeleteTable;
+    private final BlockChainInfoService blockChainInfoService;
+    private IteratorToStreamConverter<AssetDelete> assetDeleteIteratorToStreamConverter =
+        new IteratorToStreamConverter<>();
+
+    @Inject
+    public AssetDeleteServiceImpl(AssetDeleteTable assetDeleteTable,
+                                  BlockChainInfoService blockChainInfoService,
+                                  IteratorToStreamConverter<AssetDelete> assetDeleteIteratorToStreamConverter) {
+        this.assetDeleteTable = assetDeleteTable;
+        this.blockChainInfoService = blockChainInfoService;
+        if (assetDeleteIteratorToStreamConverter != null) { // for unit test only
+            this.assetDeleteIteratorToStreamConverter = assetDeleteIteratorToStreamConverter;
+        }
+    }
+
+    @Override
+    public DbIterator<AssetDelete> getAssetDeletes(long assetId, int from, int to) {
+        return assetDeleteTable.getManyBy(new DbClause.LongClause("asset_id", assetId), from, to);
+    }
+
+    @Override
+    public Stream<AssetDelete> getAssetDeletesStream(long assetId, int from, int to) {
+        return assetDeleteIteratorToStreamConverter.apply(
+            assetDeleteTable.getManyBy(new DbClause.LongClause("asset_id", assetId), from, to));
+    }
+
+    @Override
+    public DbIterator<AssetDelete> getAccountAssetDeletes(long accountId, int from, int to) {
+        return assetDeleteTable.getManyBy(new DbClause.LongClause("account_id", accountId),
+            from, to, " ORDER BY height DESC, db_id DESC ");
+    }
+
+    @Override
+    public Stream<AssetDelete> getAccountAssetDeletesStream(long accountId, int from, int to) {
+        return assetDeleteIteratorToStreamConverter.apply(
+            assetDeleteTable.getManyBy(new DbClause.LongClause("account_id", accountId),
+                from, to, " ORDER BY height DESC, db_id DESC "));
+    }
+
+    @Override
+    public DbIterator<AssetDelete> getAccountAssetDeletes(long accountId, long assetId, int from, int to) {
+        return assetDeleteTable.getManyBy(new DbClause.LongClause("account_id", accountId).and(new DbClause.LongClause("asset_id", assetId)),
+            from, to, " ORDER BY height DESC, db_id DESC ");
+    }
+
+    @Override
+    public Stream<AssetDelete> getAccountAssetDeletesStream(long accountId, long assetId, int from, int to) {
+        return assetDeleteIteratorToStreamConverter.apply(assetDeleteTable.getManyBy(
+            new DbClause.LongClause("account_id", accountId)
```

Comment:
Service should be store-independent, we aim on different store engines (sql derived tables is one of them), so that these store-specific logic usage should be avoided on service levels

Reply:
That is our common approach and it is used in all previous service components.